mobly/controllers/android_device.py

```diff
@@ -812,12 +826,16 @@ def reboot(self):
         if self.is_bootloader:
             self.fastboot.reboot()
             return
+        snippet_info = self._get_active_snippet_info()
         self.stop_services()
         self.adb.reboot()
         self.wait_for_boot_completion()
         if self.is_rootable:
             self.root_adb()
-        self.start_services(skip_sl4a=getattr(self, KEY_SKIP_SL4A, False))
+        skip_sl4a = getattr(self, KEY_SKIP_SL4A, False) or self.sl4a is None
```

Comment:
a) Do we need both conditions?
b) Does "self.sl4a is None" work if sl4a has been skipped? Wouldn't it just cause a crash saying unknown attribute sl4a?